tests/test_cli.py

```diff
@@ -288,6 +294,7 @@ def test_build_repeat_nochange(mockargs, get_catalog_fp, test_data):
         ],
         build_base_path=tempfile.TemporaryDirectory().name,  # Use pytest fixture here?
```

Comment:
Again unrelated to this PR but I think using the tempfile module like this won't tear down the temporary directory when we're done with it properly & we might want to use a fixture (I'm pretty sure you left the comment noting fixtures).

Reply:
Actually I didn't, I had duplicated the line from something someone else had done (I assume @dougiesquire), which explains why it's there.

The issue is that I often need multiple, known-to-be-different temporary directories to be able to run many of the tests correctly - is there a way to set up some sort of fixture 'factory' for this?

Reply:
Just saw your other comment about fixturing in the E2E branch, so we'll hold this over for a new issue/PR.